source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -463,8 +470,11 @@ void CacheFilter::encodeCachedResponse() {
     decoder_callbacks_->encodeHeaders(std::move(lookup_result_->headers_), end_stream,
                                       CacheResponseCodeDetails::get().ResponseFromCacheFilter);
   }
-
-  if (lookup_result_->content_length_ > 0) {
+  if (filter_state_ == FilterState::EncodeServingFromCache && is_head_request_) {
```

Comment:
Coverage is actually fine in this PR, but more generally, this filter is missing a bunch of coverage, e.g. the red in https://storage.googleapis.com/envoy-pr/a8f4dde/coverage/source/extensions/filters/http/cache/cache_filter.cc.gcov.html. 

@ekiziv  @toddmgreer @jmarantz it would be great if we can get this filter up to full coverage, doesn't seem to be huge amount of extra testing needed.

Reply:
I will address this in the followup PR! :)

Reply:
Ack, thanks!